packages/react-components/react-menu/src/components/MenuItemLink/useMenuItemLink.ts

```diff
@@ -19,6 +20,17 @@ export const useMenuItemLink_unstable = (
 ): MenuItemLinkState => {
   // casting because the root slot changes from div to a
   const baseState = useMenuItem_unstable(props as MenuItemProps, null);
+  const setOpen = useMenuContext_unstable(context => context.setOpen);
+  const open = useMenuContext_unstable(context => context.open);
+
+  const _props = { ...props };
+  _props.onClick = useEventCallback(event => {
+    if (open) {
+      setOpen(event, { open: false, type: 'menuItemClick', event });
+    }
+    props.onClick?.(event);
+  });
+
```

Comment:
`baseState` has a `root` definition that does this already https://github.com/microsoft/fluentui/blob/ec40bb1e90e55b722a57d11182c4110bfce9051e/packages/react-components/react-menu/src/components/MenuItem/useMenuItem.tsx#L74-L87, would be better to try to reuse it

Reply:
Also, we'd need to provide as well `onKeyDown` and `onMouseEnter` handlers

Reply:
Updated!